src/docfx/lib/markdown/validation/DocsValidationExtension.cs

```diff
@@ -71,6 +73,105 @@ internal static class DocsValidationExtension
             });
         }
 
+        public static bool IsInlineImage(this MarkdownObject node)
+        {
+            switch (node)
+            {
+                case Inline inline:
+                    return inline.IsInlineImage();
+                case HtmlBlock htmlBlock:
+                    return htmlBlock.IsInlineImage();
+                default:
+                    return false;
+            }
+        }
+
+        private static bool IsInlineImage(this Inline node)
+        {
+            switch (node)
+            {
+                case LinkInline linkInline when linkInline.IsImage:
+                case TripleColonInline tripleColonInline when tripleColonInline.Extension is ImageExtension:
+                case HtmlInline htmlInline when htmlInline.Tag.StartsWith("<img", StringComparison.InvariantCultureIgnoreCase):
+                    for (MarkdownObject current = node, parent = node.Parent; current != null;)
+                    {
+                        if (parent is ContainerInline containerInline)
+                        {
+                            foreach (var child in containerInline)
+                            {
+                                if (child != current && child.IsVisible())
+                                {
+                                    return true;
+                                }
+                            }
+                            current = parent;
+                            parent = containerInline.Parent;
+                        }
+                        else
+                        {
+                            return false;
+                        }
+                    }
+                    return node.GetPathToRootExclusive().Any(o => o is TableCell);
```

Comment:
If `TableCell` is taken into consideration, may as well checking for `RowBlock` and `NestedColumnBlock` because these are our custom extension to basically create a different type of table